state-and-event-handling/updating-state.md

```diff
@@ -0,0 +1,442 @@
+
+## Working With State and Re-Rendering
+
+Re-rendering a component is hard work! Firstly, setting state is _asynchronous_. Secondly, to repeat, _every time that state updates, the component re-renders_. These two facts will aid us when debugging state.
+
+### !end-callout
+
+## Updating State As Part of Event Handling
+
+The best time to update state is in response to _handling user events_.
+
+We can use our update functions inside an event handler.
+
+### Example: Increasing Like Count
+
+Let's imagine a `Post` component. It's responsible for displaying the number of likes that a post has.
+
+Each post starts with zero likes. When we click the "like" button, the number of likes increases.
+
+Let's consider the code we need to write for this:
+
+1. The `Post` component should have a piece of state, `likesCount`.`likesCount` is ideally a piece of state, and not a `prop`, because state can be modified within a component.
+1. We should also get a function to update this piece of state, `setLikesCount`.
+1. The "like" button should listen for click events, using the attribute `onClick`.
+1. When the "like" button is clicked, we should call an event handler function. We can name this function `increaseLikes`.
+1. When our component increases likes in `increaseLikes`, we should update our state using `setLikesCount`.
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from "react";
+
+const Post = () => {
+    const [likesCount, setLikesCount] = useState(0);
+    const increaseLikes = () => {
+        console.log('We\'re inside increaseLikes!');
+        setLikesCount(likesCount + 1);
+    }
+
+    return (<section>
+        <p>The number of likes is {likesCount}.</p>
+        <button onClick={increaseLikes}>Like</button>
+    </section>)
+}
+
+export default Post;
+```
+<!-- prettier-ignore-end -->
+
+When we look at our web app, our count of likes starts at zero:
+
+![An app that reads "The number of likes is 0." and a "like" button. The browser Dev Tools are open.](../assets/state-and-event-handling_updating-state_zero-likes.png)  
+_Fig. An app that reads "The number of likes is 0."_
+
+When we click it once, we see that `increaseLikes` has been invoked... and the number of likes has increased! Our state has been updated!
+
+![An app that reads "The number of likes is 1." and a "like" button. The browser Dev Tools are open. The console reads "We're inside increaseLikes!"](../assets/state-and-event-handling_updating-state_one-like.png)  
+_Fig. An app that reads "The number of likes is 1." and a "like" button. The browser Dev Tools are open. The console reads "We're inside increaseLikes!"_
+
+Clicking on the "like" button five times continues to update state!
+
+![An app that reads "The number of likes is 5." and a "like" button. The browser Dev Tools are open. The console reads "We're inside increaseLikes!"](../assets/state-and-event-handling_updating-state_five-likes.png)  
+_Fig. An app that reads "The number of likes is 5."_
```

Comment:
minor alt text updates (include the number of times the message was displayed)
update caption
```suggestion
![An app that reads "The number of likes is 5." and a "like" button. The browser Dev Tools are open. The console reads "We're inside increaseLikes!" with a count of 5 beside.](../assets/state-and-event-handling_updating-state_five-likes.png)  
_Fig. The like count is up to five, and the console message was also printed 5 times, denoted by the 5 beside it._
```